### ISSUE TITLE
[Usage Question] Issue about monitoring learning rate during training & loss function curves

### ISSUE THREAD

**Issue by Chenghan-Sun** [atomistic-machine-learning/schnetpack#497]
Dear SchNetPack development team @ktschuett @Stefaanhess ,

I've been using the dev. version of SchNetPack for a while for one of my project to train a neural network potential. The training process looks good. But I do have two important questions:

1. In my script for the training task, I imported `lr_scheduler.ReduceLROnPlateau` from `torch.optim` to monitor my learning rate during training, with some selected choices of initial learning rate, lr decay factor, and patience etc. This learning rate scheduler function was clearly used in the stable version of SchNetPack, but seems not being highlighted in the developing version. So in order to monitor the change of my learning rate, **I hope to printout the learning rate at the end of each batch, together with the training loss and val_loss, in the progress bar**. Unfortunately, I didn't be able to figure out how to save the learning rate to the log. I also clearly noticed that the stable version of SchNetPack actually had functionality in the `schnetpack.train.hook` module, by utilizing `trn.CSVHook` and enabling `self.log_learning_rate` parameter. In addition, I tried to use the `LearningRateMonitor` callback from Pytorch Lightning, but still cannot make this happen. **So my first question is, how should I monitor the change of learning rate with loss and val_loss for each epoch and each bath in the dev version?**
2. Usually after the training finished, I'd like to have a script which can plot the change of loss function value versus each epoch / batch / time to demonstrate the process of training. In the dev. version of SchNetPack, this part of analytical code seems also be removed, but this did exist in the stable version. **So my second question is, could you provide me with a simple script for printing the loss function value versus any of the progress (epoch, batch, time) factor?** 

Thank you very much for your patience and help, and any idea or suggestions will be appreciate!

Best regards,
Chenghan Sun

**Comment by Stefaanhess** [comment]
Hi @Chenghan-Sun 
sorry for the late reply!
Since we now use pytorch-lightning to do the training of our models, you could just select one of their loggers and add them to the trainer. You can find a tutorial on logging [here](https://lightning.ai/docs/pytorch/stable/extensions/logging.html). If you train the models with our `spktrain` script, you can just use one of the configs, that are provided [here](https://github.com/atomistic-machine-learning/schnetpack/tree/master/src/schnetpack/configs/logger). In general we suggest to use `tensorboard` for logging, but we also provide the configs for `csv` and `aim`. `tensorboard` basically is a graphical tool to plot exactly the progress of you training. So from my understanding, this should solve both of you questions.
If it is not installed, just use `pip install tensorboard` and add the logger the training. If you use `spktrain`, this should already be used by default. Now the logger will make a folder called `your-modeldir/logs`, where the files for logging are stored. You can open these files with `tensorboard --logdir your-modeldir/logs`. The link to the plots will be printed out in you terminal. By clicking it, the logs will appear in you browser.
Let me know, if this helps to solves the problem for you.

Best
Stefaan

**Comment by Chenghan-Sun** [comment]
Hi Stefaan,

Fantastic! This solves my issue perfectly. Thank you very much!

Best regards,
Chenghan
